src/core/Akka.TestKit/Internal/AsyncQueue.cs

```diff
@@ -122,11 +122,8 @@ public async ValueTask<T> Take(CancellationToken cancellationToken)
         /// <returns>A <see cref="List{T}"/> containing copies of the elements of the collection</returns>
         public IEnumerable<T> ToList()
         {
-            while(_collection.OutputAvailable())
-            {
-                var p = _collection.Take(); 
-                yield return p.Value;   
-            }
+            foreach (var p in _queue.ToList())
```

Comment:
Finally understood @Arkatufus's `will empty the queue.`

Reply:
This method really should return a `List<T>`